website/source/docs/providers/consul/r/prepared_query.markdown

```diff
@@ -0,0 +1,97 @@
+---
+layout: "consul"
+page_title: "Consul: consul_prepared_query"
+sidebar_current: "docs-consul-resource-prepared-query"
+description: |-
+  Allows Terraform to manage a Consul prepared query
+---
+
+# consul\_prepared\_query
+
+Allows Terraform to manage a Consul prepared query.
+
+Managing prepared queries is done using Consul's REST API. This resource is
+useful to provide a consistent and declarative way of managing prepared
+queries in your Consul cluster using Terraform.
+
+## Example Usage
+
+```
+resource "consul_prepared_query" "service-near-self" {
+    datacenter = "nyc1"
+    token = "abcd"
+    stored_token = "wxyz"
+    name = ""
+    service = "$${match(1)}"
```

Comment:
I thought about that, and figured if that did happen at some point, we could start doing things like having, for example, `node` and `service` at the top level be mutually exclusive. We could also document certain blocks, like `failover`, as being only applicable for certain PQ types. I think it works as long as we don't have fields of the same name being different types, which would seem very odd.